plenum/server/consensus/instance_change_service.py

```diff
@@ -0,0 +1,136 @@
+from typing import Callable
+
+from plenum.common.config_util import getConfig
+from plenum.common.constants import NODE_STATUS_DB_LABEL, VIEW_CHANGE_PREFIX
+from plenum.common.event_bus import InternalBus, ExternalBus
+from plenum.common.messages.internal_messages import VoteForViewChange, PreNeedViewChange, NewViewAccepted
+from plenum.common.messages.node_messages import InstanceChange
+from plenum.common.metrics_collector import MetricsCollector, NullMetricsCollector
+from plenum.common.router import Subscription
+from plenum.common.stashing_router import StashingRouter, DISCARD
+from plenum.common.timer import TimerService
+from plenum.server.consensus.consensus_shared_data import ConsensusSharedData
+from plenum.server.database_manager import DatabaseManager
+from plenum.server.suspicion_codes import Suspicions, Suspicion
+from plenum.server.view_change.instance_change_provider import InstanceChangeProvider
+from stp_core.common.log import getlogger
+
+
+class InstanceChangeService:
```

Comment:
Maybe it's better (taking into account upcoming changes for ReplicaRemoving) to call this class `ViewChangeTriggerService`?

Reply:
Agree, for some reason I couldn't come up with good enough name, so I kept `InstanceChangeService` for a while. I like `ViewChangeTriggerService` much better.